src/components/NewChatForm.tsx

```diff
@@ -2,28 +2,51 @@ import { useMessagingStore } from "../store/messaging.store";
 import { useWalletStore } from "../store/wallet.store";
 import React, { useState, useCallback, useEffect, useRef } from "react";
 import { kaspaToSompi, sompiToKaspaString } from "kaspa-wasm";
+import { KaspaAddress } from "./KaspaAddress";
 import styles from "../components/NewChatForm.module.css";
 
 interface NewChatFormProps {
   onClose: () => void;
+  prefilledRecipient?: string;
+  prefilledResolvedAddress?: string;
+  prefilledDomainId?: string;
 }
 
-export const NewChatForm: React.FC<NewChatFormProps> = ({ onClose }) => {
-  const [recipientAddress, setRecipientAddress] = useState("");
+export const NewChatForm: React.FC<NewChatFormProps> = ({
+  onClose,
+  prefilledRecipient = "",
+  prefilledResolvedAddress = "",
+  prefilledDomainId = "",
+}) => {
+  const [recipientAddress, setRecipientAddress] = useState(prefilledRecipient);
```

Comment:
recipient address should be renamed to `recipientValue` since it can contains either an address or a KNS domain